cookbooks/swift/recipes/rings.rb

```diff
@@ -62,14 +65,25 @@
     cwd "/etc/swift"
   end
   (1..num_disks).each do |i|
-    j = ((i - 1) % node['nodes']) + 1
+    n_idx = ((i - 1) % node['nodes']) + 1
     z = ((i - 1) % node['zones']) + 1
     r = ((z - 1) % node['regions']) + 1
-    execute "#{service}.builder-add-sdb#{i}" do
+    dev = "sdb#{i}"
+    ip = "127.0.0.1"
+    port = "60#{n_idx}0"
+    if node['servers_per_port'] > 0 then
+      ip = "127.0.0.#{n_idx}"
+
+      # Range ports per disk per node from 60j6 - 60j9
+      # NOTE: this only supports DISKS <= 4 * NODES
+      p = 5 + (i / Float(node['nodes'])).ceil.to_int
+      port = "60#{n_idx}#{p}"
```

Comment:
so right now we have account & container's on ports `60#{n_idx}1` and `60#{n_idx}2` - in the non per-port case objects use `60#{n_idx}0`.

Maybe in the next go ground if  `servers_per_port > 0` - objects should just move to `"6#{n_idx}%02d" % i` - should work up to 99 disks and keep sdbXX matching with 6?XX.